### DISCUSSION TITLE
Help with installation

### DISCUSSION THREAD

**Discussion by ShiranGlasser** [dov/giv#66] [Q&A]
Hi, I'm trying to install giv on my ubuntu.
I ran into 2 issues in the beginning of the process:
- when I'm trying to install gob2 i gev this error:
<img width="371" alt="image" src="https://github.com/dov/giv/assets/96018625/fabd4167-6982-4974-b45e-0cd8d0c6c48b">

- when I'm trying to run the command `./autogen.sh --prefix=/usr/local; make -j 8 install` in libplis I'm receiving this error:
<img width="546" alt="image" src="https://github.com/dov/giv/assets/96018625/e31739dc-7018-47d3-94be-a95818e10c1b">

even though I installed https://ftp.gnu.org/pub/gnu/libtool/libtool-1.0.tar.gz

I would appreciate if you could help me with this issues. Thank you,
Shiran - shiranglasser10@gmail.com

**Comment by dov** [comment]
Hi @ShiranGlasser . Thanks for the bug report. I have now amended the INSTALL.md file and also tested all the commands on clean VM with Ubuntu 23.10 . In INSTALL.md  have also listed the necessary installation packages for Ubuntu. Please test again and let me know of any problems.

Good luck and dash to Shimon.

**Reply by ShiranGlasser** [reply]
Hi, @dov. I now get another error after running the `scons -j 8` command. the error:
Package dcmtk was not found in the pkg-config search path.
Perhaps you should add the directory containing `dcmtk.pc'
to the PKG_CONFIG_PATH environment variable
No package 'dcmtk' found
OSError: 'env PKG_CONFIG_PATH=/usr/local/lib/pkgconfig:/usr/local/lib64/pkgconfig pkg-config --cflags --libs dcmtk plis'   

How can I solve this issue? Thank you very much for the fast answer.

**Reply by dov** [reply]
@ShiranGlasser Did you do `apt-get install libdcmtk-dev` as I wrote in the INSTALL.md file?

If it doesn't work, you can simply search for the string "dcmtk" in the SConscript files and comment out the build of the dcmtk plugin.

Hope this helps.

**Reply by ShiranGlasser** [reply]
@dov 
I did install libdcmtk-dev and still got the error.. eventually what helped   with the building was the command: `scons -j 8 nodicom=1 -fPIC` 
Now the error on the final stage of installing is;
scons -j 8 nodicom=1 -fPIC install:

scons: Reading SConscript files ...
scons: warning: Calling missing SConscript without error is deprecated.
Transition by adding must_exist=0 to SConscript calls.
Missing SConscript 'PIC'
File "/usr/bin/scons", line 204, in <module>
scons: done reading SConscript files.
scons: Building targets ...
scons: *** Do not know how to make File target `install' (/home/shirangl/DevTools/giv/install).  Stop.
scons: building terminated because of errors.

**Reply by dov** [reply]
@ShiranGlasser I used your request to some long overdue cleanups. Here is a list of the issues that I have done:

* I have replaced all scons compilations with meson. 
* I have updated INSTALL.md to reflect this
* I have created a VM with Ubuntu 20.04.4 and fixed some minor issues that stopped compilation
* Verified that giv compiles and runs on Ubuntu 20.04.4

Please remove the previous directories of libplis, and giv, and create a clean clone, and then follow the instructions in INSTALL.md .

**Reply by dov** [reply]
I have now simplified the building even further, by making the dependencies (gob2 and libplis) into meson subprojects. This means that you can fully ignore them, and all that is needed to build giv is to run meson as follows:

```
meson setup build
cd build
ninja
ninja install
```

INSTALL.md has been updated to reflect this. Please let me know if there are still any problems.

**Reply by ShiranGlasser** [reply]
@dov Thank you so much for your availability responses! 
unfortunately, I have a new error, hope it's the last one..
`meson.build:18:0: ERROR: Subproject directory not found and gob2.wrap file not found`

**Reply by dov** [reply]
Hi @ShiranGlasser . After I sent my last message, I discovered that there are still some issues. I promise to look into them in the next few days. My goal is that the installation should be trivial. Thanks for giving me the incentive. 🙂

**Reply by dov** [reply]
I had forgotten to push some files. I have fixed it now. Please pull and try again.

חג שמח!

**Reply by ShiranGlasser** [reply]
@dov  Installed! thank you!!

**Reply by dov** [reply]
🥳
